### ISSUE TITLE
Allow closed ports in “Port” object

### ISSUE THREAD

**Issue by deadbits** [ReconJSON/ReconJSON#30]
The current spec defines that closed ports should not be included in the JSON. For the purposes of OSINT and fingerprinting, a closed port is as useful to know about as an open port. This can allow you to rescan a host at a later time and compare the previous scan with the latest to see which ports have changed/opened.

For example, if you have a Host and it’s running Nginx on port 80, port 25 is closed, port 22 is open running OpenSSHD — then you scan it again a month later and port 25 is now open, port 80 and port 22 are closed, it’s very possible that the Host isn’t being operated by the same owner or it is no longer being used for the same purpose. This is valuable information to know when trying to assess what a Host is (C2 server, compromised host, etc) and can help in aging out indicators from you feed/database. 

## Expected Behavior
Allow for Ports to contain the standard nmap-style designations of: open, closed, filtered, or unfiltered. 

## Current Behavior
Schema only allows for open ports to be added. 

## Steps to Reproduce

1.
2.
3.
4.

**Comment by Edznux** [comment]
Hi,

Couldn't it be assumed that non-stored port are closed ?
I would agree with filtered port, we might want to add `{"filtered": true}` to the service.

I would think using the `active` field in the Service might be enough for your use case.

**Comment by bcoles** [comment]
> assumed

Non-stored ports may not have been scanned. `{"Schrodinger": true}`

> we might want to add `{"filtered": true}` to the service.

I, for one, look forward to 65,535 TCP port objects and 65,535 UDP port objects for every host.

**Comment by deadbits** [comment]
I'm with this person^

**Comment by 0xdevalias** [comment]
My quick /2c here is that allowing for it in the spec would give the most flexible usecases for downstream workflows. And if we use well defined states for closed/filtered, then it’s easy to filter out results you don’t care about at your storage/persistence layer, etc.

**Comment by deadbits** [comment]
My thoughts exactly. I don't see any real downside to including it.

IMO, modeling the field directly to Nmap's port values would be an easy way to go, as essentially everyone in infosec has used Nmap and is used to their understandings of values 

```The state is either open, filtered, closed, or unfiltered.
Open means that an application on the target machine is listening for connections/packets on that port. 
Filtered means that a firewall, filter, or other network obstacle is blocking the port so that Nmap cannot tell whether it is open or closed. 
Closed ports have no application listening on them, though they could open up at any time. 
Ports are classified as unfiltered when they are responsive to Nmap's probes, but Nmap cannot determine whether they are open or closed. 
Nmap reports the state combinations open|filtered and closed|filtered when it cannot determine which of the two states describe a port```

Reference: 
https://wiki.onap.org/plugins/servlet/mobile?contentId=35520753#content/view/35520753
